New landing page

fstoqnov-iohk opened the pull request:
Landing page 
 - features ✅
 - content abstraction ✅
 - animations ✅
 - dependencies ✅
 - header (kept the default docusaurus burger menu as it's unclear currently what content we'll be keeping for the header) ✅

Docs pages - ✅

Note - For consistency I have implemented TailwindCss (I noticed that it is already used in the repo in other places) - it's primarily used in styling the landing page and some parts of swizzled components that I had to update.
I have maintained the docusaurus styling method for all other components & pages

fstoqnov-iohk committed:
filip(feat): docs landing page - responsive implementation

fstoqnov-iohk committed:
filip(feat): docs landing page - animations and remaining content

ch1bo commented on docs/src/css/custom.css:
```diff
@@ -4,6 +4,10 @@
 
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Seems like we are using tailwind now. Can you update the documentation above to mention this and how this relates to the "infima" CSS framework? Maybe one is used for the custom landing page while the other is used by docusaurus?

fstoqnov-iohk replied:
Sure thing - I'll provide some documentation on it

ch1bo replied:
Leaving this unresolved as I can't seem to find that docs?

ch1bo replied:
We'll live without it.

ch1bo commented on docs/docusaurus.config.js:
```diff
@@ -243,11 +279,7 @@ const config = {
-          Built with Docusaurus on ${docsMetadataJson.site.lastUpdatedAt}
-          </small>
-          `,
+        copyright: `© 2024`,
```
Are we legally required to put © somewhere on the page? The project's copyright and license notices would be available on the repository too (but are outdated).

fstoqnov-iohk replied:
As far as I'm aware it's a legal requirement and this was part of the approved design/copy for this piece of work

fstoqnov-iohk committed:
filip(feat): docs landing page - apply feedback and fixes

fstoqnov-iohk committed:
filip(feat): non-landing page design implementation (to account for the landing page changes)

leandroiohk committed:
Tweak landing page styles and fonts

ffakenz commented on docs/docs/homepage/case-studies.ts:
```diff
@@ -0,0 +1,11 @@
+  content: translate({
+    id: "homepage.caseStudy.content",
+    message:
+      "From enabling micropayments with near-zero fees and instant processing to boosting the performanceof decentralized finance (DeFi) applications and enhancing real-time gaming experiences, Hydra can enhance user engagement across various sectors. Discover how Hydra’s cost-efficient, scalable, andlow-latency transactions can address real-world problems and support innovative solutions on the  ...
```
```suggestion
      "From enabling micropayments with near-zero fees and instant processing to boosting the performanceof decentralized finance (DeFi) applications and enhancing real-time gaming experiences, Hydra can enhance user engagement across various sectors. Discover how Hydra’s cost-efficient, scalable, and low-latency transactions can address real-world problems and support innovative solutions on the Cardano platform.",
```

ffakenz commented on docs/docusaurus.config.js:
```diff
@@ -14,10 +14,10 @@ const editUrl = "https://github.com/cardano-scaling/hydra/tree/master/docs";
   url: "https://input-output-hk.github.io",
   baseUrl: "/head-protocol/",
-  onBrokenLinks: "throw",
+  onBrokenLinks: "warn",
```
why did we relax this?

ch1bo replied:
Yes. Must: keep the `throw`

ffakenz commented on docs/docusaurus.config.js:
```diff
@@ -136,37 +147,49 @@ const config = {
     ({
+      colorMode: {
+        defaultMode: "light",
+        disableSwitch: true,
```
why?

ianhanssoniohk replied:
New design doesn't have light/dark colour modes, dev did enquire if this has been approved to removed and told that it was.

ch1bo replied:
Can't recall that approval, but fine by me - hereby approved. We can re-add it later / enable it only for the docs pages if we want.

ffakenz commented:
From my point of view, when building the docs, I see there are a lot of unnecessary traces:
```
{
  disabled: false,
  shouldRender: false,
  toggle: [Function (anonymous)],
  shown: false
}
```
which i wonder if it is possible to get rid of.

ffakenz commented on docs/src/theme/Footer/Layout/index.js:
```diff
@@ -0,0 +1,66 @@
+                  : "text-black"
+              )}
+            >
+              Lorem Ipsum is simply dummy text of the printing{" "}
```
TODO

ianhanssoniohk replied:
I've asked the PM if they have content for this as we haven't been provided it yet.

ch1bo replied:
We decided to not need an additional span here.

ffakenz commented:
Noticed all links are broken, do you think we can get them back?

ffakenz commented:
🙏 Please, I beg you: 

can we find better images for `docs/static/img/case-studies.png` and `docs/static/img/case-studies-mobile.png`? 

I find them current ones completely misleading.

ffakenz commented:
This new landing page looks sick!! :fire::fire::fire:

Great Job! :clap:

ianhanssoniohk commented:
> 🙏 Please, I beg you:
> 
> can we find better images for `docs/static/img/case-studies.png` and `docs/static/img/case-studies-mobile.png`?
> 
> I find them current ones completely misleading.

I'll check with PM / Design.

leandroiohk committed:
Tweak footer

ianhanssoniohk commented:
@leandroiohk We'll need to optimise the images and videos properly included here statically, will cover tomorrow.

leandroiohk committed:
Fix console warnings

leandroiohk committed:
Tweak "How it Works" section's styles and reveal animation

leandroiohk committed:
Fix space between word and comma on animated text

ch1bo commented on docs/docusaurus.config.js:
```diff
@@ -182,7 +205,7 @@ const config = {
               },
               {
-                label: "Architectural Decision Records",
+                label: "Architectural Decision",
```
```suggestion
                label: "Architectural Decision Records",
```

ch1bo commented on docs/src/components/homepage/Carousel/Coursel.tsx:
Typo on the filename?

ch1bo commented on docs/static/hydra-docs-landing-graphic.png:
Must: fix this image

Layer 1 and Layer 2 are swapped!

ianhanssoniohk replied:
If you could specify any changes here to the designer in Jira as we take whatever is in Figma.
https://input-output.atlassian.net/browse/MK-4572

ch1bo reviewed (changes requested):
The file `docs/static/hydra-docs-landing-graphic.png` is wrong, see comment. Where are the sources for this to fix?

My second major point: I see MANY vendored components from the docusaurus theme. I can't imagine that all of them needed changing and maybe the whole theme was vendored and modified in some parts only? Which of those have been modified and how? My ask: _can we keep the vendoring to a minimum?_

leandroiohk committed:
Tweak internal pages

leandroiohk committed:
Fix useWindowSize filename

leandroiohk committed:
Fix carousel moving to the last slide on page load (tablet)

leandroiohk committed:
Fix carousel filename

leandroiohk committed:
Replace case studies images

leandroiohk committed:
Replace "how it works" images

leandroiohk committed:
Update video and image assets with optimised versions

leandroiohk committed:
Tweak footer and case studies text

ch1bo reviewed (changes requested):
Another issue with the current site is that it does not `yarn build`: https://github.com/cardano-scaling/hydra/actions/runs/10303762271/job/28522603240?pr=1549

Judging from the error, it seems to trip over react component hooks in server side rendering.

ianhanssoniohk commented:
> Another issue with the current site is that it does not `yarn build`: https://github.com/cardano-scaling/hydra/actions/runs/10303762271/job/28522603240?pr=1549
> 
> Judging from the error, it seems to trip over react component hooks in server side rendering.

Yeah we have a fix for this just working through some final amends and getting the build to play nice.

leandroiohk committed:
Adjust headings on markdown areas

leandroiohk committed:
Tweak mobile menu

leandroiohk committed:
Revert onBrokenLinks back to "throw"

ch1bo committed:
Remove unmodified vendored theme components

These components have been "swizzled", but were not modified to the
upstream sources. Confirmed by running

for f in $(find src/theme/ -type d); do
  yarn swizzle @docusaurus/theme-classic -e --danger "${f#src/theme/}"
done

And then looking at components which were not semantically modified.

ch1bo commented on docs/docs/homepage/features.tsx:
```diff
@@ -0,0 +1,107 @@
+    }),
+    tagLine: translate({
+      id: "homepage.featureList.lowLatency.perk",
+      message: "Fast and cheap transactions",
```
Can we just keep these "taglines" empty? They are _currently redundant_ and although they look good, we likely don't need a tagline on such a small description above:

![image](https://github.com/user-attachments/assets/433b7abb-294b-4bb3-a487-56f3b2255295)

ch1bo commented on docs/src/pages/index.tsx:
```diff
@@ -0,0 +1,33 @@
+        <main>
+          <Features />
+          <div className="pageContainer">
+            <AnimatedText />
```
Should: fix the animated text. It runs off-screen:

![image](https://github.com/user-attachments/assets/4bed3ba2-9cd3-4868-a418-4f44b88e52cd)

leandroiohk committed:
Replace "why hydra head" image and tweak reveal animation

ch1bo commented on docs/docs/homepage/how-it-works.ts:
This is not really about "How it works" so the naming of the content / component over in `src/components/Homepage` is off.. but I'd say we can fix that later still.

leandroiohk replied:
Renamed to Why Hydra Head

ianhanssoniohk committed:
Replace useLocation with context to determine if on landing page to resolve static build issues.

ianhanssoniohk committed:
Remove swizzled breadcrumb and resolve minor static build styling differences

ianhanssoniohk committed:
Cleanup styling on ADR / Blog post layout

leandroiohk committed:
Remove gap on top of the hero and tweak faq headings and logo hover

leandroiohk committed:
Additional docs styles

leandroiohk committed:
Use .markdown instead of .theme-doc-markdown to apply styles to the ADR pages as well

leandroiohk committed:
Tweak GitHub logo in the footer

leandroiohk committed:
Fix animated text on firefox

leandroiohk committed:
Rename "How It Works" files to "Why Hydra Head"

noonio commented:
> > Another issue with the current site is that it does not `yarn build`: https://github.com/cardano-scaling/hydra/actions/runs/10303762271/job/28522603240?pr=1549
> > Judging from the error, it seems to trip over react component hooks in server side rendering.
> 
> Yeah we have a fix for this just working through some final amends and getting the build to play nice.

following this, it would be great to update the readme for the docs with build instructions, as I don't know if you've fixed this and i also get errors with both `yarn build` and `npm i`


yarn

```
...
[INFO] [en] Creating an optimized production build...
[ERROR] Unable to build website for locale en.
[ERROR] Error: Cannot find module 'tailwindcss'
Require stack:
- /home/noon/dev/w/hydra/docs/docusaurus.config.js
- /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/server/config.js
- /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/server/index.js
- /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/commands/build.js
- /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:121:18)
    at Object.configurePostCss (/home/noon/dev/w/hydra/docs/docusaurus.config.js:71:39)
    at overridePostCssOptions (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/webpack/utils.js:151:44)
    at /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/webpack/utils.js:159:36
    at Array.forEach (<anonymous>)
    at overridePostCssOptions (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/webpack/utils.js:159:18)
    at /home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/webpack/utils.js:162:45
[INFO] Docusaurus version: 2.4.1
Node version: v18.17.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

-- edit:

you need to run `yarn` alone, and _then_ `yarn build`, apparently. 🙄 ; so this one works, but npm still not. (should it?!)


-- double edit

i still get a `yarn build` failure; but a different one now

```
[ERROR] Unable to build website for locale en.
[ERROR] Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /head-protocol/benchmarks/profiling:
   -> linking to /head-protocol/haddock/hydra-plutus/Hydra-Contract-Commit.html
   -> linking to /head-protocol/haddock/hydra-plutus/Hydra-Contract-Head.html
   -> linking to /head-protocol/haddock/hydra-node/Hydra-Ledger-Cardano-Evaluate.html#v:prepareTxScripts (resolved as: /head-protocol/haddock/hydra-node/Hydra-Ledger-Cardano-Evaluate.html)

- On source page path = /head-protocol/audit-guidelines:
   -> linking to /head-protocol/haddock/hydra-node/tests/Hydra-Chain-Direct-Contract-Mutation.html

    at throwError (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/logger/lib/index.js:76:11)
    at handleBrokenLinks (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/server/brokenLinks.js:153:47)
    at async buildLocale (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/commands/build.js:186:5)
    at async tryToBuildLocale (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/commands/build.js:41:20)
    at async mapAsyncSequential (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/utils/lib/jsUtils.js:34:24)
    at async Command.build (/home/noon/dev/w/hydra/docs/node_modules/@docusaurus/core/lib/commands/build.js:76:21)
```

npm
```
02:36 pm φ docs (filip/feat/docs-landing-page +) npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: hydra-head-protocol-docs@0.0.0
npm ERR! Found: react@18.3.1
npm ERR! node_modules/react
npm ERR!   react@"^18.3.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.4 || ^17.0.0" from @docusaurus/core@2.4.3
npm ERR! node_modules/@docusaurus/core
npm ERR!   @docusaurus/core@"^2.4.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/noon/.npm/_logs/2024-08-09T13_36_20_573Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /home/noon/.npm/_logs/2024-08-09T13_36_20_573Z-debug-0.log
```

noonio commented:
Thanks for the work on this :)

Can we keep this in Draft mode until it's ready for review? Or, alternatively, have a list of items that need to be checked off so we know when it's ready to look at? Thanks ! :)

ianhanssoniohk commented:
> noonio

The broken links failing on a static build locally I believe would be due to other dependencies which in the pipeline are pulled in through other actions. To my knowledge this isn't something changed in this PR.

noonio commented:
> > noonio
> 
> The broken links failing on a static build locally I believe would be due to other dependencies which in the pipeline are pulled in through other actions. To my knowledge this isn't something changed in this PR.

True; it fails on master as well; in the exact same way.

😕

leandroiohk committed:
Add staggered animation to feature grid

leandroiohk committed:
Update carousel copy

ianhanssoniohk commented on docs/src/components/homepage/Carousel/ResponsiveCarousel.tsx:
@leandroiohk I know this is from previously but don't really like how this desktop and mobile carousel are entirely separate components when they are largely the same with some variations would be good to aim to consolidate and only apply required differences for mobile.

ch1bo replied:
Going to merge it now. @leandroiohk or @fstoqnov-iohk feel free to address this in a follow-up PR

leandroiohk committed:
Remove tag lines from features

ch1bo commented on docs/static/hydra-docs-landing-graphic.png:
This is better now. Not exactly as I want it, but good enough to merge.

ch1bo commented on docs/docusaurus.config.js:
```diff
@@ -136,37 +147,49 @@ const config = {
+          //   to: "/core-concepts",
+          //   label: "Core Concepts",
+          //   position: "right",
+          // },
```
Should: keep it until we find a better place for these

fstoqnov-iohk replied:
@ch1bo Are you happy with us to merge it in if you're happy with it - it looks good on our end as well - or shall we await other approval before we do the merge to prod ?

ch1bo replied:
Let's merge the whole thing and we'll see it pushed to https://hydra.family/head-protocol/unstable. Then we can resolve some last things still (we would ping you if we need help) and publish it in the next days to normal https://hydra.family/head-protocol

ch1bo reviewed (approved):
Landing page looks great.

Minor inconsistencies we can resolve in follow-up PRs.

Thanks!

fstoqnov-iohk committed:
filip(chore): unused swizzled components cleanup
